packages/ee/billing/handlers/subscriptionCreatedOrUpdated.ts

```diff
@@ -51,127 +25,52 @@ export const handleSubscriptionUpdatedOrCreated = async (event: Stripe.Event) =>
   }
 
   const team = await getTeam(teamId);
-  if (!team) throw new Error("Team not found.");
-  let updatedFeatures = team.billing.features;
-
-  let scheduledSubscriptions: Stripe.SubscriptionSchedule[] = [];
-  if (stripeSubscriptionObject.cancel_at_period_end) {
-    const allScheduledSubscriptions = await stripe.subscriptionSchedules.list({
-      customer: team.billing.stripeCustomerId as string,
-    });
-    scheduledSubscriptions = allScheduledSubscriptions.data.filter(
-      (scheduledSub) => scheduledSub.status === "not_started"
-    );
+  if (!team) throw new Error("Team not found");
+
+  const product = await stripe.products.retrieve(
+    stripeSubscriptionObject.items.data[0].price.product as string
+  );
+  if (!product) throw new Error("Product not found");
+
+  let updatedBilling = team.billing;
+  let responses = product.metadata.responses as unknown as number;
```

Comment:
We shouldn't type cast these as numbers, they won't actually be converted to numbers, but the typescript compiler will still be treating them as numbers which could lead to runtime errors. Please use `parseInt` here to actually convert these to numbers 🙏